chain/paloma/collision/zero_collision_strategy.go

```diff
@@ -0,0 +1,143 @@
+package collision
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"hash/fnv"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/palomachain/pigeon/util/slice"
+	log "github.com/sirupsen/logrus"
+)
+
+// Zero collision strategy ensures that all pigeons are using the same way of determinig if they can execute a message or not.
+// That way we can distribute the jobs without having to write data to paloma so that paloma could do it, as it could be slow.
+// Of course, paloma does not care about the strategy pigeons are using. Somebody could rewrite pigeon to do send all the messages,
+// but then they would be "fighting" with paloma and they (and other pigeons) would spend gas unceccecary.
+// A "bad" actor would only be doing a job for us :).
+
+const (
```

Comment:
So this is a randomization of the selection for the queue? Could this not be attacked by malicious pigeons that want to setup pigeons for denial of service?

Reply:
The worst thing that can happen is that a single pigeon sends all the messages from a queue to a target chain, and that pigeon would be doing a job for us anyway.